packages/frontend/src/pages/public/SignInPage.tsx

```diff
@@ -37,13 +112,111 @@ const SignInPage: React.FC<SignInPageProps> = () => {
+          <div className="flex flex-col pb-10 px-14">
+            <StyledFirebaseAuth
+              className="w-full"
+              uiConfig={uiConfig}
+              firebaseAuth={auth}
+            />
+            {/* the text with line before and after */}
+            <div className="flex items-center py-5 ">
+              <div className="flex-grow border-t border-black"></div>
+              <div className="px-2 pb-1">
+                <Text weight="semibold">
+                  {isLogin ? 'Login with E-mail' : 'Or sign up with E-mail'}
+                </Text>
+              </div>
+              <div className="flex-grow border-t border-black"></div>
+            </div>
+            {/* input fields group starts */}
+            {isLogin ? (
+              <></>
+            ) : (
+              <>
+                <Input
+                  onChange={(v) => setName(v.target.value)}
+                  contentLeft={<UserIcon style={{ height: '100%' }} />}
+                  placeholder="Name"
+                />
+                <Spacer y={0.5} />
+              </>
+            )}
+            <Input
+              onChange={(v) => setEmail(v.target.value)}
+              contentLeft={<MailIcon style={{ height: '100%' }} />}
+              placeholder="Your email"
+            />
+            <Spacer y={0.5} />
+            <Input.Password
+              onChange={(v) => setPassword(v.target.value)}
+              contentLeft={<LockOpenIcon style={{ height: '100%' }} />}
+              placeholder="Your password"
+            />
+            <Spacer y={0.5} />
+            {/* Show forgot password on login, and show another password input field on sign up */}
+            {isLogin ? (
+              <div>
+                <div className="flex justify-end">
+                  <Button size="xs" light>
+                    <Text size={12}>Forgot Your Password?</Text>
+                  </Button>
+                </div>
+                <Spacer y={0.75} />
+              </div>
+            ) : (
+              <Input.Password
+                onChange={(v) => setConfirmPassword(v.target.value)}
+                contentLeft={<LockOpenIcon style={{ height: '100%' }} />}
+                placeholder="Confirm password"
+              />
+            )}
+            {/* Button for sigup/login */}
+            <Button
+              className="my-5"
+              style={{
+                backgroundColor: '#2596A4',
+              }}
+              onClick={() =>
+                isLogin ? validateLoginInput() : validateSignUpInput()
+              }
+            >
+              {isLogin ? 'Login' : 'Sign Up'}
+            </Button>
+            <div className="flex justify-center">
+              {/* Button for switch login/signup */}
+              <Button size="xs" light onClick={() => setIsLogin(!isLogin)}>
+                <Text size={12}>
+                  {isLogin
+                    ? "Don't have an account? Click here to sign up"
```

Comment:
You'll have to check this works, not sure if Text will accept jsx children.

```suggestion
                    ? <>{"Don't have an account? "}<span className="text-teal-500">{"Click here to sign up"}</span></>
```

Reply:
Yup this works.